File: tabulation/best_sum/python_impl/best_sum.py
def best_sum(target, numbers):
    table = [None]*(target+1)
    table[0] = []
    for index, element in enumerate(table):
        if element != None:
            for number in numbers:
                if (index+number) <= target:
                    a = [*table[index], number]
                    if not table[index+number] or len(table[index+number]) > len(a):
                        table[index+number] = a

    return table[target]


print(best_sum(7, [5, 3, 4, 7]))
print(best_sum(8, [2, 3, 5]))
print(best_sum(8, [1, 4, 5]))
print(best_sum(100, [1, 2, 5,25]))

File: tabulation/best_sum/js_impl/bestSum.js
const bestSum = (target, numbers) => {
    const table = Array(target + 1).fill(null);
    table[0] = [];
    for (let i = 0; i < table.length; i++) {
        if (table[i] !== null) {
            for (let element of numbers) {
                if ((i + element) <= target) {
                    const combination = [...table[i], element];
                    if (!table[i + element] || table[i + element].length > combination.length) {
                        table[i + element] = combination;
                    }
                }

            }
        }
    }
    return table[target];
}


console.log(bestSum(7, [5, 3, 4, 7]))
console.log(bestSum(8, [2, 3, 5]))
console.log(bestSum(8, [1, 4, 5]))
console.log(bestSum(100, [1, 2, 5, 25]))